lib/collection/src/shards/replica_set/snapshots.rs

```diff
@@ -70,8 +70,16 @@ impl ShardReplicaSet {
         Ok(())
     }
 
-    /// Returns if local shard was recovered from path
-    pub async fn restore_local_replica_from(&self, replica_path: &Path) -> CollectionResult<bool> {
+    /// # Safety
+    ///
+    /// This function is *not* cancel-safe!
```

Comment:
I am still a bit confused about the term *cancel-safe* while also adding a cancellation token option 😅. 

Not requesting changes for it, but I kinda wish the term was a Future being *drop-safe*, if only it wasn't an already established term

Reply:
Convention is to call it 'cancel safety', so I guess we better stick with it.

It's commonly documented like this, so I guess it's best to follow that exact pattern, sometimes with additional context:

```rust
/// # Cancel safety
///
/// This method is cancel safe.
```

This is on the Tokio codebase: 
![image](https://github.com/qdrant/qdrant/assets/856222/72562dbc-151d-497d-b9bb-0c162f806458)

Reply:
I'll do some Vim magic on it now to make them all follow the same pattern.